iroh-gateway/src/constants.rs

```diff
@@ -1,17 +1,31 @@
+use axum::http::{header::HeaderName, HeaderValue};
+
 // Headers
-pub const HEADER_X_IPFS_PATH: &str = "X-Ipfs-Path";
-pub const HEADER_X_CONTENT_TYPE_OPTIONS: &str = "X-Content-Type-Options";
-pub const HEADER_X_TRACE_ID: &str = "X-Trace-Id";
+pub static HEADER_X_IPFS_PATH: HeaderName = HeaderName::from_static("x-ipfs-path");
```

Comment:
why static?

Reply:
Clippy is complaining because there are mutable internals in HeaderName. Not sure how to move forward with that.

Reply:
See https://github.com/n0-computer/iroh/pull/36#discussion_r872125102